### PR TITLE
dev/core#1077 Fixes making report listing actions links hookable

### PR DESCRIPTION
**Pull request by sushantpaste**
Overview
----------------------------------------
Making report action link on report listing page hookable.
Before
----------------------------------------
Not able to change action links with civicrm links hook.

After
----------------------------------------
Able to change action links with civicrm links hook.

Comments
----------------------------------------
More details: https://lab.civicrm.org/dev/core/issues/1077

### CODE REVIEW

**Comment by mattwire** [comment]
This looks good in principal.  It's the right place for the hook.  Two things:
1. We need a documentation PR here https://github.com/civicrm/civicrm-dev-docs
2. `getActionLinks()` seems to return an array with slightly different from "standard" keys and this means the hook would also behave differently in this case.  I'm wondering whether we should fix this at the same time as adding the hook? - it would just mean updating the template I think to read the "new" keys.

Eg. <pre>
      CRM_Core_Action::UPDATE => [
        'name' => ts('Settings'),
        'url' => 'civicrm/group',
        'qs' => 'reset=1&action=update&id=%%id%%',
        'title' => ts('Edit Group'),
      ],
</pre>

That would normally be passed through the `CRM_Core_Action::formLink()` function to be displayed.

File: CRM/Report/Page/InstanceList.php [hunk 1]
```diff
@@ -282,7 +282,12 @@ protected function getActionLinks($instanceID, $className) {
         'confirm_message' => ts('Are you sure you want delete this report? This action cannot be undone.'),
       ];
     }
-
+    //Call link hook.
+    CRM_Utils_Hook::links('view.report.links',
+      NULL,
```
Comment (pradpnayak) [inline CRM/Report/Page/InstanceList.php]: Any reason for setting objectName to NULL?

File: CRM/Report/Page/InstanceList.php [hunk 2]
```diff
@@ -282,7 +282,12 @@ protected function getActionLinks($instanceID, $className) {
...
+      CRM_Core_DAO::$_nullObject,
```
Comment (pradpnayak) [inline CRM/Report/Page/InstanceList.php]: report instance id?

File: CRM/Report/Page/InstanceList.php [hunk 3]
```diff
@@ -282,7 +282,12 @@ protected function getActionLinks($instanceID, $className) {
...
+      $actions
+    );
```
Comment (pradpnayak) [inline CRM/Report/Page/InstanceList.php]: Should we need any other useful info to pass as 6th param?

**Comment by colemanw** [comment]
@sushantpaste this is waiting for your response

**Comment by yashodha** [comment]
@sushantpaste awaiting your response.

**Comment by sushantpaste** [comment]
@colemanw @pradpnayak @yashodha I don't think we need report instance id or abject as its something which user needs for every instance ,like removing some links with hooks for particular report instance. What's your suggestion on this?

**Comment by yashodha** [comment]
@sushantpaste IMO we should expose instance id and classname in the hook as is consistent with other places
https://github.com/civicrm/civicrm-core/blob/master/CRM/Contact/Page/View.php#L375

**Comment by sushantpaste** [comment]
Closing this PR Here is the new PR with changes: https://github.com/civicrm/civicrm-core/pull/15074

**Pull request closed without merging**